firmware/application/apps/ui_mictx.cpp

```diff
@@ -657,8 +662,10 @@ MicTXView::~MicTXView() {
     audio::input::stop();
     transmitter_model.set_target_frequency(tx_frequency);
     transmitter_model.disable();
-    if (rx_enabled)  // Also turn off audio rx if enabled
+    if (rx_enabled) {  // Also turn off both (audio rx if enabled, and disable  mic_loop to HP)
         rxaudio(false);
+        audio::input::loopback_mic_to_hp_disable();  // Leave Mic audio off in the main menu (as original audio path, otherwise we had No audio in next "Audio App")
```

Comment:
Shouldn't we turn off the mic loopback all of the time when exiting the Mic app (versus only when rx_enabled)?

Reply:
Well, we had exactly the same thought . And I think you are correct.
Theoretically it should be done all the time , not only at the exit of the receiver .
It is implicit done inside audio:input::stop() , but it does not work ...

Moreover , at the exit of that Mic App ,  the first instruction is 
audio::input::stop();

void stop() {
    i2s::i2s0::rx_stop();
    audio_codec->microphone_disable();
}

And actually inside microphone_disable() , we already are disabling that loopback ,
   
 void microphone_disable() override {
        microphone_mute(true);
        microphone_to_HP_disable();
    }

It is the same contents as loopback_disable, 

void loopback_mic_to_hp_disable() {
    audio_codec->microphone_to_HP_disable();
}

Anyway , adding loopback_mic_to_hp_disable() on top all the time , did not solve it .
But adding it inside the receiver, after stopping the flag of rxaudio, it works...

I still do not know why , but it seems only to work well inside that conditional if (receiver) .
Maybe I am missing something ..., but in that position, it solves exactly our problem. Maybe the key is 
send that command , once we stopped the receiver....
Then I think we should better keep as it is . 
How do you feel ?

Reply:
I need to study this a bit more.  I never like it when firmware is behaving differently than we expect.  :-)